project/AkkaDependency.scala

```diff
@@ -8,64 +8,60 @@ import sbt._
 import Keys._
 
 object AkkaDependency {
-  // Property semantics:
-  // If akka.sources is set, then the given URI will override everything else
-  // else if akka version is "master", then a source dependency to git://github.com/akka/akka.git#master will be used
-  // else if akka version is "default", then the hard coded default will be used (jenkins doesn't allow empty values for config axis)
-  // else if akka.version is anything else, then the given version will be used
 
-  // Updated only when needed
-  // https://doc.akka.io//docs/akka/current/project/downstream-upgrade-strategy.html
-  val defaultAkkaVersion = "2.5.30"
-  val akkaVersion =
-    System.getProperty("akka.http.build.akka.version", defaultAkkaVersion) match {
-      case "default" => defaultAkkaVersion
-      case x => x
-    }
-
-  // Needs to be a URI like git://github.com/akka/akka.git#master or file:///xyz/akka
-  val akkaSourceDependencyUri = {
-    val fallback =
-      akkaVersion match {
-        case "master" => "git://github.com/akka/akka.git#master"
-        case "release-2.5" => "git://github.com/akka/akka.git#release-2.5"
-        case x => ""
-      }
+  sealed trait Akka
+  case class Artifact(version: String) extends Akka
+  case class Sources(uri: String) extends Akka
 
-    System.getProperty("akka.sources", fallback)
+  def akkaDependency(defaultVersion: String): Akka = {
+    Option(System.getProperty("akka.sources")) match {
+      case Some(akkaSources) =>
+        Sources(akkaSources)
+      case None =>
+        Option(System.getProperty("akka.http.build.akka.version")) match {
+          case Some("master") => Sources("git://github.com/akka/akka.git#master")
+          case Some("release-2.5") => Sources("git://github.com/akka/akka.git#release-2.5")
+          case Some("default") => Artifact(defaultVersion)
+          case Some(other) => Artifact(other)
+          case None => Artifact(defaultVersion)
+        }
+    }
   }
-  val shouldUseSourceDependency = akkaSourceDependencyUri != ""
-
-  val akkaRepository = {
-    // as a little hacky side effect also disable aggregation of samples
-    System.setProperty("akka.build.aggregateSamples", "false")
+  // Default version updated only when needed, https://doc.akka.io//docs/akka/current/project/downstream-upgrade-strategy.html
+  val default = akkaDependency(defaultVersion = "2.5.30")
+  val docs = akkaDependency(defaultVersion = "2.6.4")
 
-    uri(akkaSourceDependencyUri)
+  val akkaVersion: String = default match {
```

Comment:
Has the value changed to before? Have you checked where this is used? It probably doesn't matter but would make sense to check.